currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -0,0 +1,313 @@
+# Copyright 2009-2016 Camptocamp
+# Copyright 2010 Akretion
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCurrencyRateProvider(models.Model):
+    _name = 'res.currency.rate.provider'
+    _description = 'Currency Rates Provider'
+    _inherit = ['mail.thread']
```

Comment:
I don't think we need this overhead here.

Reply:
Don't you think it's important to keep track of changes on provider? Source/currencies/etc?

Reply:
Not too much in fact. This is transient information for the aim of having the rates.

Reply:
I would accept it if errors are logged in this chatter and are displayed in an easy way. Maybe you can put even an activity for an error.

Reply:
So remove tracking and add error logging? How to do the latter?

Reply:
This can be done with manual calls to `post` method (not exact name) when an error happens. In previous versions, this was done in an ugly Text field if I don't remember bad.